scripts/zkai_attention_kv_d64_four_head_seq32_fused_softmax_table_native_gate.py

```
@@ -0,0 +1,1225 @@
+        os.close(fd)
+
+
+def ensure_parent_dir_without_symlinks(path: pathlib.Path, label: str) -> None:
+    candidate = path if path.is_absolute() else pathlib.Path.cwd() / path
+    reject_symlinked_output_path(candidate, label)
+    try:
+        candidate.parent.mkdir(parents=True, exist_ok=True)
+    except OSError as err:
+        raise AttentionKvD64FourHeadSeq32FusedSoftmaxTableGateError(
+            f"failed to create parent directory {candidate.parent} for {label}: {err}"
+        ) from err
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)